### PR TITLE
feat: adding BSC support

### PR DESCRIPTION
**Pull request by bertux**
## Description of the changes
Adding BSC network support to add BANANA token needed by APE Finance.

These contracts have been deployed on BSC testnet :

- EthereumProxy:                0x27c60BE17e853c47A9F1d280B05365f483c2dFAF
- ERC20FeeProxy:                0x0DfbEe143b42B41eFC5A6F87bFD1fFC78c2f0aC9
- ChainlinkConversionPath:      0xEEc4790306C43DC00cebbE4D0c36Fadf8634B533
- ERC20Conversion:              0xf0f49873C50765239F6f9534Ba13c4fe16eD5f2E
- ERC20SwapToPay:               0x75740D9b5cA3BCCb356CA7f0D0dB71aBE427a835
- ERC20SwapToConversion:        0x1d6B06C6f7adFd9314BD4C58a6D306261113a1D4

Testing ERC20 support with tokens received from faucet :

- DAI: 0xec5dcb5dbf4b114c9d0f65bccab49ec54f6a0867

### CODE REVIEW

**Comment by vrolland** [comment]
It misses the deployment of smartcontracts on the new blockchain and the new artifacts related to them (in the package smartcontracts)

**Comment by bertux** [comment]
> It misses the deployment of smartcontracts on the new blockchain and the new artifacts related to them (in the package smartcontracts)

That's why it's still a draft ;-) We have deployed the smartcontracts just now on testnet so my next stetp is to update in the package smartcontracts the artifacts related to them.

**Comment by vrolland** [comment]
> That's why it's still a draft ;-) 

oh! I guess you requested a code review from me by mistake. :)

File: packages/smart-contracts/src/lib/artifacts/ERC20SwapToPay/index.ts [hunk 1]
```diff
@@ -20,6 +20,10 @@ export const erc20SwapToPayArtifact = new ContractArtifact<ERC20SwapToPay>(
           address: '0xb674e3d228e631594D8fd4BF947E1811288bf836',
           creationBlockNumber: 7363204,
         },
+        bsctest: {
```
Comment (vrolland) [inline packages/smart-contracts/src/lib/artifacts/ERC20SwapToPay/index.ts:23]: Same here.
You added twice the same address for two different versions. you should add the address only for the relevant version.

File: packages/smart-contracts/src/lib/artifacts/ERC20FeeProxy/index.ts [hunk 2]
```diff
@@ -45,6 +45,10 @@ export const erc20FeeProxyArtifact = new ContractArtifact<ERC20FeeProxy>(
           address: '0x0DfbEe143b42B41eFC5A6F87bFD1fFC78c2f0aC9',
           creationBlockNumber: 18326896
         },
+        bsctest: {
```
Comment (vrolland) [inline packages/smart-contracts/src/lib/artifacts/ERC20FeeProxy/index.ts:48]: You added twice the same address for two different versions. you should add the address only for the relevant version.

**Review by benjlevesque — APPROVED** [review]

File: packages/payment-detection/src/provider.ts [hunk 3]
```diff
@@ -36,6 +36,7 @@ const networkRpcs: Record<string, string> = {
   matic: 'https://rpc-mainnet.matic.network/',
   fantom: 'https://rpcapi.fantom.network',
   fuse: 'https://rpc.fuse.io',
+  bsc: 'https://bsc-dataseed1.binance.org',
```
Comment (vrolland) [inline packages/payment-detection/src/provider.ts:39]: No need for `bsctest` provider?

File: packages/advanced-logic/src/extensions/payment-network/erc20/fee-proxy-contract.ts [hunk 4]
```diff
@@ -23,6 +23,8 @@ export default class Erc20FeeProxyPaymentNetwork<
       'celo',
       'alfajores',
       'fuse',
+      'bsc',
```
Comment (vrolland) [inline packages/advanced-logic/src/extensions/payment-network/erc20/fee-proxy-contract.ts:26]: only `bsctest` is supported as the deployment on `bsc` have not been done.

**Review by vrolland — CHANGES_REQUESTED** [review]
- Update of supported network & currencies (in advanced logic) missing for `EthereumProxy` and `ERC20Conversion`
- Consider renaming the PR to be more descriptive.

File: packages/smart-contracts/src/lib/artifacts/ERC20SwapToPay/index.ts (hunk 1 above)
Comment (bertux) [reply packages/smart-contracts/src/lib/artifacts/ERC20SwapToPay/index.ts]: thank you, done by keeping it only in the latest version.

File: packages/smart-contracts/src/lib/artifacts/ERC20FeeProxy/index.ts (hunk 2 above)
Comment (bertux) [reply packages/smart-contracts/src/lib/artifacts/ERC20FeeProxy/index.ts]: thank you, done by keeping it only in the latest version.

File: packages/payment-detection/src/provider.ts (hunk 3 above)
Comment (bertux) [reply packages/payment-detection/src/provider.ts]: I've replaced bsc by bsctest for this first test.

File: packages/advanced-logic/src/extensions/payment-network/erc20/fee-proxy-contract.ts (hunk 4 above)
Comment (bertux) [reply packages/advanced-logic/src/extensions/payment-network/erc20/fee-proxy-contract.ts]: I've replaced bsc by bsctest for this first test.

**Review by vrolland — APPROVED** [review]
Nice work. Looks good to me :)

**Pull request merged**